### PR TITLE
feat(load-ind): Add tests for loading indicator

### PR DESCRIPTION
**Pull request by udhadpa**
Hello,
Please review my PR for loading indicator. Mobile and desktop tests covered.
@wyseguyonline @JasonLantz @StommePoes @mhomolak @umahaea 
Thanks,

### CODE REVIEW

File: src/test/java/origamiV2Tests/LoadingIndicatorTest.java [hunk 1]
```diff
@@ -0,0 +1,469 @@
...
+        if (!isElementPresent) {
+            log.info("Loading indicator " + type + " is not displayed");
+        }
+        Assert.assertTrue(isElementPresent);
+    }
+
+    @Test(testName = "Press Tab on Button Test", dataProvider = "Click on Button Test Data", groups = {"desktop-regression"})
+    private void TabOnButtonTest(String type, By button, By elem, String[] propsPropertiesList) throws Exception {
+        if (browser.equals("safari")) {
+            throw new SkipException("Safari does not support this action");
```
Comment (StommePoes) [inline src/test/java/origamiV2Tests/LoadingIndicatorTest.java]: I thought safari supported tabbing to buttons, just not tabbing to links?

File: src/test/java/origamiV2Tests/LoadingIndicatorTest.java [hunk 2]
```diff
@@ -0,0 +1,469 @@
...
+
+            //build the final props json object with all the attributes
+            propsJsonDetailObject = new JsonObject();
+            for (Map.Entry<String, String> entry : propsConfigMap.entrySet()) {
+                propsJsonDetailObject.addProperty(entry.getKey(), entry.getValue().toString());
+            }
+            /*************************************************************************************/
+            //prepare the map for detail properties
+            detailMap = new LinkedHashMap<String, String>();
+            for (i = 0; i < (detailsPropertiesList.length - 1); i = i + 2) {
```
Comment (StommePoes) [inline src/test/java/origamiV2Tests/LoadingIndicatorTest.java]: another place where we can i+=2

File: src/test/java/origamiV2Tests/LoadingIndicatorTest.java (hunk 1 above)
Comment (udhadpa) [inline src/test/java/origamiV2Tests/LoadingIndicatorTest.java]: Hi Mallory, I checked it again. Safari does not support tabbing on buttons too, my tests failed
https://travis-ci.org/Pearson-Higher-Ed/ux-test-platform/builds/256134040

Comment (StommePoes) [inline src/test/java/origamiV2Tests/LoadingIndicatorTest.java]: ok good to know! I thought this was related to that setting required to make Safari/Mac tab to links (by default on a web page you can tab to buttons and maybe form inputs but nothing else unless you change the setting).

File: src/test/java/origamiV2Tests/LoadingIndicatorTest.java [hunk 3]
```diff
@@ -0,0 +1,469 @@
...
+    Map<String, JsonObject> dataTextMap = null;
+    Map<String, JsonObject> propsMap = null;
+    Map<String, JsonObject> dataMap = null;
+    Map<String, String> propsConfigMap = null;
+    Map<String, String> dataTextConfigMap = null;
+    final static Logger log = Logger.getLogger(origamiV2Tests.LoadingIndicatorTest.class.getName());
+    LoadingIndicatorPageObjects loadIndicatorPgObj = null;
+
+    @BeforeClass(alwaysRun = true)
+    private void InputsTestBeforeClass() {
```
Comment (umahaea) [inline src/test/java/origamiV2Tests/LoadingIndicatorTest.java]: beforeclass method to be renamed..

File: src/test/java/origamiV2Tests/LoadingIndicatorTest.java [hunk 4]
```diff
@@ -0,0 +1,469 @@
...
+        color = commonUtils.getCSSValue(elem, "color", "mobile");
+        isChipText = commonUtils.assertValue(chipText, setChipText, "Chip text value set does not match the value in Data Provider");
+        isColor = commonUtils.assertCSSProperties("color", color, new String[]{commonUtils.hex2Rgb("#ffffff"), commonUtils.hex2RgbWithoutTransparency("#ffffff")});
+        if (!isColor) {
+            log.info("Color of the chip text is not as per spec, actual: " + color);
+        }
+        Assert.assertTrue(isChipText && isColor);
+    }
+
+    private String buildJSONObjectDetailConfig(String type, String[] detailsPropertiesList, String[] propsPropertiesList, String[] dataTextPropertiesList) throws IOException {
```
Comment (umahaea) [inline src/test/java/origamiV2Tests/LoadingIndicatorTest.java]: Bravo!!! for writing this complicated logic!

**Comment by udhadpa** [comment]
Got approvals, merging this merge

**Pull request merged**